pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -25,42 +25,30 @@ const (
 	functionNameEnvVar = "AWS_LAMBDA_FUNCTION_NAME"
 )
 
-// executionStartInfo is saved information from when an execution span was started
-type executionStartInfo struct {
+// ExecutionStartInfo is saved information from when an execution span was started
+type ExecutionStartInfo struct {
```

Comment:
Like I mentioned before, this is a good class to pass by value, (instead of pointer), because it is a plain old data object, without any pointers/slices/maps inside. Making a copy guarantees the underlying data won't change on another thread. Go shallow copies values, so if it had any pointer, slice or map fields, that guarantee would go out the window.